integration-tests/tests/test_cca_verify_challenge.tavern.yaml

```diff
@@ -0,0 +1,99 @@
+test_name: cca-verify-challenge
```

Comment:
Does this test checks negative case also, meaning Supplied `nonce` is not part of the token ?

Reply:
No, because that would be checked in the policy that is not part of the code being tested, but rather the test. What is being tested here is that the nonce associated with the session is now available to the policy (this also serves as an example and an answers one of the questions you asked above -- why is it useful of the nonce to be available to the policy), not the fact that the nonce in the token is being validated (it isn't outside this test).